File: java-tests/com/hounshell/kotlin-sonnet/java/types/TypeReferenceTests.java
package com.hounshell.kotlin_sonnet.java.types;

import static com.google.common.truth.Truth.assertThat;

import com.hounshell.kotlin_sonnet.types.KotlinRealTypeReference;

import org.junit.Test;

public class TypeReferenceTests {
    @Test
    public void typeReferenceByName() {
        assertThat(new KotlinRealTypeReference("foo.Bar").asDeclaration()).isEqualTo("foo.Bar");
    }

    @Test
    public void TypeReferenceByJavaClass() {
        assertThat(new KotlinRealTypeReference( String.class).asDeclaration()).isEqualTo( "String");
    }
}

File: java-tests/com/hounshell/kotlin-sonnet/kotlin/types/TypeReferenceTests.kt
package com.hounshell.kotlin_sonnet.kotlin.types

import com.google.common.truth.Truth.assertThat
import com.hounshell.kotlin_sonnet.types.RealTypeReference
import com.hounshell.kotlin_sonnet.types.TypeReference

import org.junit.Test

class TypeReferenceTests {
    @Test
    fun `TypeReference by name`() {
        assertThat(RealTypeReference("foo", "Bar").asDeclaration()).isEqualTo("foo.Bar")
    }

    @Test
    fun `TypeReference by java class`() {
        assertThat(RealTypeReference(String::class.java).asDeclaration()).isEqualTo("String")
    }
}
